src/extension/background-script/state.ts

```diff
@@ -90,7 +96,7 @@ const state = createState<State>((set, get) => ({
   lock: async () => {
     const connector = get().connector;
     if (connector) {
-      connector.unload();
+      await connector.unload();
```

Comment:
unload is async

this should not be in this PR 🙈

Reply:
I think these minor improvements are ok here

Reply:
not necessarily minor... one word with potential big implications.